astroid/nodes/scoped_nodes.py

```diff
@@ -1734,7 +1734,11 @@ def infer_yield_result(self, context=None):
         :returns: What the function yields
         :rtype: iterable(NodeNG or Uninferable) or None
         """
-        for yield_ in self.nodes_of_class(node_classes.Yield):
+        for (
+            yield_
+        ) in self.nodes_of_class(  # pylint: disable=not-an-iterable # See https://github.com/PyCQA/astroid/issues/1015
```

Comment:
Just realized that we would need to add this to every call in `pylint` as well 😕
The issue should probably be fixed before we can merge it.

Reply:
I did a `pip instal -e local_astroid_folder` and succeeded in running the pre-commit to test for this.

The problem comes from `overload`s where the `...` is on a separate line (I think).
We don't have those in pylint (yet), so I think for the time-being we are fine.

You might want to check this though by doing `pip -e` as well, just to be sure!

Reply:
Just tested it as well and you're right. At the moment `pylint` doesn't complain. The reason however isn't that we don't use `overloads`. It's actually that `node` in `visit_xxx(self, node)` cannot be inferred and thus no error is emitted. In contrast, just doing something like this will easily produce one:
```py
def func():
    node = nodes.For()
    for n in node.nodes_of_class(nodes.Name):  # not-an-iterable
        pass
```

Reply:
Didn't we just add typing to all `visit_xxx(self, node)` calls in `pylint`? Or is that not enough to help `mypy`?

I saw you just assigned somebody to #1015. I think we should block this until that issue has been fixed, right?

Reply:
Except from some corner cases, type annotations **aren't** used to infer values in pylint. That is different in `mypy`.

>  I think we should block this until that issue has been fixed, right?

Since this doesn't actually cause any issues with `pylint`, I would say we can merge it now. It doesn't necessarily need to wait for #1015.